### PR TITLE
[CA-1307] A11y: Workspace pages

### PR DESCRIPTION
**Pull request by andrewmarcus**
This PR attempts to address accessibility issues on all of the pages related to an individual workspace, enumerated here: https://broadworkbench.atlassian.net/browse/CA-1307

To get these pages fully compliant, this PR depends on the updates to Tooltips from #2525, which was split out from this branch and should be merged first.

Although this PR contains a large number of changes, the main theme is better semantics. Links which open popup modals now use `aria-haspopup="dialog"`, collapsible sections now properly link to the region which they control, and Autocomplete widgets now link properly to a label.

In addition, I made significant semantic improvements to the navigation section of the Data page on workspaces. The sidebar is now a nested `role="list"` within a `role="navigation`" region, with `aria-current` used to indicate which table is active, and it no longer contains buttons within buttons.

### CODE REVIEW

File: src/components/common.js
Comment (andrewmarcus) [inline src/components/common.js]: I snuck this in here so I could start to use the attribute without waiting for #2525 to get merged and without causing DOM warnings. May possible cause a merge conflict once that branch is merged, but it'll be easy to resolve.

File: src/components/common.js
Comment (andrewmarcus) [inline src/components/common.js]: #2525 is necessary in order to resolve the original issue identified by aXe where this button has no `aria-label`.

File: src/components/data/EntitiesContent.js
Comment (andrewmarcus) [inline src/components/data/EntitiesContent.js]: #2525 will set `aria-hidden` to false automatically if `aria-label` is present. No need to add that here and then remove it once branch is merged...

File: src/components/data/EntitiesContent.js
Comment (andrewmarcus) [inline src/components/data/EntitiesContent.js:381]: This announces the current selection status to screen readers whenever it changes. `role="status"` uses `aria-live="polite"` by default.

File: src/style.css
Comment (andrewmarcus) [inline src/style.css:165]: I missed these colors in the first update to the Prism colors. This time I found a workflow that had tokens from this group, and ANDI flagged the contrast again.

File: src/components/data/LocalVariablesContent.js
Comment (marctalbott) [inline src/components/data/LocalVariablesContent.js]: Is this the right aria label? Completely fine if the answer is yes, but it seems like they're generally shorter and I'd expect it to be more along the lines of "Edit Type"

File: src/components/data/EntitiesContent.js
Comment (s-rubenstein) [inline src/components/data/EntitiesContent.js]: Was the whitespace change here intentional? (newline)

File: src/components/table.js
Comment (s-rubenstein) [inline src/components/table.js]: Out of curiosity, how come we don't need the `aria-label`?

File: src/pages/workspaces/workspace/Data.js
Comment (s-rubenstein) [inline src/pages/workspaces/workspace/Data.js:64]: I'm surprised that undefined works here, given some of the other things about tooltip. Should this be `unnamed entity` or something? (That way we guarantee a label)

File: src/pages/workspaces/workspace/JobHistory.js
Comment (s-rubenstein) [inline src/pages/workspaces/workspace/JobHistory.js]: It looks like there's a conditional removed here. What is the goal of that change?

File: src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js
Comment (s-rubenstein) [inline src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js]: Nit: In general, I prefer `marginLeft` to `marginRight`, because it means that an element is more concerned with its own positioning, rather than the positioning of other elements around it. 

Also, if possible, I would recommend doing this with rem rather than em

Comment (s-rubenstein) [inline src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js]: Nit: Because there are more styles being applied to these than just flex now, I would say we should probably pull out to a variable.

File: src/pages/workspaces/workspace/workflows/WorkflowView.js
Comment (s-rubenstein) [inline src/pages/workspaces/workspace/workflows/WorkflowView.js]: Nit: RM newline

**Review by marctalbott — APPROVED** [review]

File: src/components/data/EntitiesContent.js
Comment (andrewmarcus) [reply src/components/data/EntitiesContent.js]: I had added other stuff, then removed it with the Tooltips enhancements. I'll revert the whitespace

File: src/components/data/LocalVariablesContent.js
Comment (andrewmarcus) [reply src/components/data/LocalVariablesContent.js]: Fair point. I was trying to describe that it applies a data type to other content, but maybe "data type" would be a reasonable enough label

File: src/pages/workspaces/workspace/Data.js
Comment (andrewmarcus) [reply src/pages/workspaces/workspace/Data.js:64]: Technically, in this instance the tooltip is only necessary in situations where the `noWrapEllipsis` hides some of the content. Otherwise it's basically just duplicative of what the user already sees. 

As far as I have seen, the only cases where this widget displays its children rather than an entity name is for the Reference Data, and the words "Workspace Data" and "Files" at the end. Since these don't extend beyond 280px, a tooltip is technically unnecessary for them.

File: src/pages/workspaces/workspace/JobHistory.js
Comment (andrewmarcus) [reply src/pages/workspaces/workspace/JobHistory.js]: Ah. I moved the conditional logic into the `statusCell` function since I needed the content to be nested rather than parallel. I'll remove the `Fragment` since there's now only one child here

File: src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js
Comment (andrewmarcus) [reply src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js]: Hmm, the problem had to do with wrapping in this section on narrow windows. If one or two of the links bumped down to the next line, they were indented by 1rem compared to the first line. But looking at it now, I'm not sure these fixes solve that either.

Since it's not technically linked to accessibility, perhaps I'll just revert this part and let the Broad team figure out how you want to handle it.
Screen Shot 2021-06-04 at 11 37 01 PM

Comment (andrewmarcus) [reply src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js]: OBE: I reverted to closer to how they were

File: src/components/table.js
Comment (andrewmarcus) [reply src/components/table.js]: Oh. Right. I pulled this out because #2525 will make it redundant; the tooltip will be used as a label once that branch is merged. At one time those changes were in this branch...

I can restore that line for now, to keep the two PRs entirely independent. Or leave it like this assuming they'll be merged at roughly the same time.  Your choice. :)

Comment (andrewmarcus) [reply src/components/table.js]: Nevermind. The change is already in that PR. I'll revert it here

File: src/pages/workspaces/workspace/Workflows.js
Comment (s-rubenstein) [inline src/pages/workspaces/workspace/Workflows.js]: Nit (existing, so feel free to ignore): We should probably have consistent capitalization on this, either `Agora` instead of `agora` or `terra` instead of `Terra`

**Review by s-rubenstein — APPROVED** [review]
This looks reasonable to me! Thanks for the changes.

File: src/pages/workspaces/workspace/Workflows.js
Comment (andrewmarcus) [reply src/pages/workspaces/workspace/Workflows.js]: This is just the human-readable name. Maybe use `Utils.normalizeLabel()`?

Comment (s-rubenstein) [reply src/pages/workspaces/workspace/Workflows.js]: That sounds like exactly the right method to use!

File: src/components/data/EntitiesContent.js
Comment (dmohs) [inline src/components/data/EntitiesContent.js]: I believe we are placing 'aria-...' attributes first in the list of properties for consistency.

**Review by dmohs — APPROVED** [review]

**Pull request merged**
